Core/include/Acts/AmbiguityResolution/GreedyAmbiguityResolution.ipp

```diff
@@ -0,0 +1,74 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/AmbiguityResolution/GreedyAmbiguityResolution.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t, typename source_link_hash_t,
+          typename source_link_equality_t>
+void GreedyAmbiguityResolution::computeInitialState(
+    const TrackContainer<track_container_t, traj_t, holder_t>& tracks,
+    State& state, source_link_hash_t&& sourceLinkHash,
+    source_link_equality_t&& sourceLinkEquality) const {
+  auto measurementIndexMap =
+      std::unordered_map<SourceLink, std::size_t, source_link_hash_t,
+                         source_link_equality_t>(0, sourceLinkHash,
+                                                 sourceLinkEquality);
+
+  for (const auto& track : tracks) {
+    auto trajState = Acts::MultiTrajectoryHelpers::trajectoryState(
```

Comment:
Track should have the number of measurements already I think, so there's probably no need to recompute anymore?